app/src/main/java/com/example/ncov19traking/NCoVRecyclerAdapter.kt

```diff
@@ -88,9 +94,28 @@ class NCoVRecyclerAdapter : RecyclerView.Adapter<NCoVRecyclerAdapter.ViewHolder>
 
     }
 
-    fun calculatePercentage(universeNumber : Int, fieldNumber : Int) : String{
-       return fieldNumber.times(100).div(universeNumber.toFloat()).format(2)
+    private fun calculatePercentage(universeNumber: Int, fieldNumber: Int): String {
+        return fieldNumber.times(100).div(universeNumber.toFloat()).format(2)
     }
 
     fun Float.format(digits: Int) = "%.${digits}f".format(this)
+
+    private fun sortByActiveCases() {
+        dataset.sortByDescending { it.active }
+    }
+
+    private fun sortAlphabetically() {
+        dataset.sortBy { it.country }
+    }
+
+    fun changeSort(isSortedByCases: Boolean) {
+        if (isSortedByCases) {
+            sortByActiveCases()
+            notifyDataSetChanged()
+        } else {
+            sortAlphabetically()
+            notifyDataSetChanged()
```

Comment:
you are repeating code with this `notifyDataSetChanged()` you can leave it out of the condition.

Reply:
thats true, done